README


This archive contains the distribution Task-BeLike-DOY,
version 0.11:

  modules I use

This software is copyright (c) 2013 by Jesse Luehrs.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Revision history for Task-BeLike-DOY

0.11  2013-07-26
      - couple more deps

0.10  2013-06-26
      - switch to Carp::Reply, add Clipboard

0.09  2013-06-08
      - switch to reply, switch away from ack, add Unicode::Tussle

0.08  2012-11-07
      - remove some broken modules

0.07  2012-11-07
      - more updates

0.06  2012-03-21
      - more updates

0.05  2012-02-13
      - Task::SDK::Moose no longer exists

0.04  2011-05-16
      - more updates

0.03  2011-01-26
      - some updates

0.02  2010-06-13
      - add a few more things
      - convert to more in depth dzil stuff

0.01  2010-05-02
      - Initial release


lib/Task/BeLike/DOY.pm
package Task::BeLike::DOY;
BEGIN {
  $Task::BeLike::DOY::AUTHORITY = 'cpan:DOY';
}
{
  $Task::BeLike::DOY::VERSION = '0.11';
}
#ABSTRACT: modules I use


1;

__END__

=pod

=head1 NAME

Task::BeLike::DOY - modules I use

=head1 VERSION

version 0.11

=head1 TASK CONTENTS

=head2 Useful apps

=head3 L<App::Nopaste> 0.20

Version 0.20 required because: has Codepeek pastebin

=head3 L<App::ph>

=head3 L<App::Termcast> 0.06

=head3 L<App::Ttyrec>

=head3 L<LastFM::Export>

=head3 L<Term::TtyRec::Plus> 0.09

Version 0.09 required because: for ttyplay and ttytime

=head3 L<Unicode::Tussle>

=head2 Development tools

=head3 L<App::ForkProve>

=head3 L<Carp::Always::Color> 0.03

=head3 L<Carp::Reply>

=head3 L<circular::require>

=head3 L<Data::Dump>

=head3 L<Devel::Cover>

=head3 L<Devel::FindRef>

=head3 L<Devel::NYTProf> 3.00

=head3 L<Devel::STDERR::Indent>

=head3 L<Devel::Trace>

=head3 L<Dist::Zilla> 4.00

=head3 L<Dist::Zilla::PluginBundle::DOY> 0.09

=head3 L<ExtUtils::MakeMaker::Dist::Zilla::Develop>

=head3 L<lib::require::all>

=head3 L<Reply>

=head3 L<Test::Pretty>

=head3 L<Test::Without::Module>

=head2 Useful optional deps for things

=head3 L<autodie>

=head3 L<Archive::Zip>

=head3 L<B::Keywords>

=head3 L<Devel::PartialDump>

=head3 L<HTTP::Parser::XS>

=head3 L<JSON>

=head3 L<LWP>

=head3 L<LWP::Protocol::https>

=head3 L<Pod::Coverage::TrustPod>

=head3 L<Proc::InvokeEditor>

=head3 L<Term::ReadLine::Gnu>

=head3 L<Test::Pod>

=head3 L<Test::Pod::Coverage>

=head3 L<YAML>

=head2 CPAN interaction

=head3 L<App::cpanminus> 1.5007

Version 1.5007 required because: for build.log prompting

=head3 L<App::cpanoutdated> 0.05

Version 0.05 required because: for offline mode

=head3 L<App::pmuninstall>

=head3 L<App::PAUSE::Comaint>

=head3 L<CPAN::Mini>

=head3 L<Module::Which>

=head3 L<Pod::Cpandoc>

=head2 Modules I use a lot

=head3 L<Bread::Board::Declare>

=head3 L<DateTime>

=head3 L<Moose> 2.0400

=head3 L<namespace::autoclean>

=head3 L<Path::Class> 0.25

Version 0.25 required because: for ->traverse

=head3 L<Plack>

=head3 L<Scope::Guard>

=head3 L<Term::ProgressBar>

=head3 L<Test::Fatal>

=head3 L<Test::More> 0.88

=head3 L<Test::Requires>

=head2 Dependencies for my non-CPAN apps

=head3 L<Clipboard>

=head3 L<File::pushd>

=head3 L<IO::Pty::Easy> 0.08

=head3 L<JSON>

=head3 L<WWW::Mechanize>

=begin taskweaver_parsing_is_broken




=end taskweaver_parsing_is_broken

=head1 AUTHOR

Jesse Luehrs <doy at tozt dot net>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Jesse Luehrs.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "modules I use",
  "AUTHOR" => "Jesse Luehrs <doy at tozt dot net>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Task-BeLike-DOY",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Task::BeLike::DOY",
  "PREREQ_PM" => {
    "App::ForkProve" => 0,
    "App::Nopaste" => "0.20",
    "App::PAUSE::Comaint" => 0,
    "App::Termcast" => "0.06",
    "App::Ttyrec" => 0,
    "App::cpanminus" => "1.5007",
    "App::cpanoutdated" => "0.05",
    "App::ph" => 0,
    "App::pmuninstall" => 0,
    "Archive::Zip" => 0,
    "B::Keywords" => 0,
    "Bread::Board::Declare" => 0,
    "CPAN::Mini" => 0,
    "Carp::Always::Color" => "0.03",
    "Carp::Reply" => 0,
    "Clipboard" => 0,
    "Data::Dump" => 0,
    "DateTime" => 0,
    "Devel::Cover" => 0,
    "Devel::FindRef" => 0,
    "Devel::NYTProf" => "3.00",
    "Devel::PartialDump" => 0,
    "Devel::STDERR::Indent" => 0,
    "Devel::Trace" => 0,
    "Dist::Zilla" => "4.00",
    "Dist::Zilla::PluginBundle::DOY" => "0.09",
    "ExtUtils::MakeMaker::Dist::Zilla::Develop" => 0,
    "File::pushd" => 0,
    "HTTP::Parser::XS" => 0,
    "IO::Pty::Easy" => "0.08",
    "JSON" => 0,
    "LWP" => 0,
    "LWP::Protocol::https" => 0,
    "LastFM::Export" => 0,
    "Module::Which" => 0,
    "Moose" => "2.0400",
    "Path::Class" => "0.25",
    "Plack" => 0,
    "Pod::Coverage::TrustPod" => 0,
    "Pod::Cpandoc" => 0,
    "Proc::InvokeEditor" => 0,
    "Reply" => 0,
    "Scope::Guard" => 0,
    "Term::ProgressBar" => 0,
    "Term::ReadLine::Gnu" => 0,
    "Term::TtyRec::Plus" => "0.09",
    "Test::Fatal" => 0,
    "Test::More" => "0.88",
    "Test::Pod" => 0,
    "Test::Pod::Coverage" => 0,
    "Test::Pretty" => 0,
    "Test::Requires" => 0,
    "Test::Without::Module" => 0,
    "Unicode::Tussle" => 0,
    "WWW::Mechanize" => 0,
    "YAML" => 0,
    "autodie" => 0,
    "circular::require" => 0,
    "lib::require::all" => 0,
    "namespace::autoclean" => 0
  },
  "TEST_REQUIRES" => {
    "Test::More" => "0.88"
  },
  "VERSION" => "0.11",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



